src/Loess.jl

```diff
@@ -135,56 +135,48 @@ end
 # Returns:
 #   A length n' vector of predicted response values.
 #
-function predict(model::LoessModel{T}, z::T) where T <: AbstractFloat
-    predict(model, T[z])
-end
-
 
-function predict(model::LoessModel{T}, zs::AbstractVector{T}) where T <: AbstractFloat
-    m = size(model.xs, 2)
+# univariate model, prediction at single point
+function predict(model::LoessModel{false,T,V,M}, z::T) where {T <: AbstractFloat, V <: AbstractVector{T}, M <: AbstractMatrix{T}}
+    adjacent_verts = traverse(model.kdtree, T[z])
+    @assert(length(adjacent_verts) == 2)
+    v₁, v₂ = adjacent_verts[1][1], adjacent_verts[2][1]
+    interpolate(model,z,v₁,v₂)
+end
 
-    # in the univariate case, interpret a non-singleton zs as vector of
-    # ponits, not one point
-    if m == 1 && length(zs) > 1
-        return predict(model, reshape(zs, (length(zs), 1)))
-    end
+# univariate model, prediction at multiple points
+function predict(model::LoessModel{false,T,V,M}, zs::AbstractVector{T}) where {T <: AbstractFloat, V <: AbstractVector{T}, M <: AbstractMatrix{T}}
```

Comment:
```suggestion
function predict(model::LoessModel{false}, zs::AbstractVector)
```
The element types aren't used in the body so the restrictions shouldn't be necessary.